Zain-Bajwa/country_crawler
# feat: add 4 approaches to parse country information

### **Testing Instructions**
- Clone the repository `https://github.com/Zain-Bajwa/country_crawler.git`
- Create virtual environment
```bash
$ python -m venv .venv
$ source .venv/bin/activate
```
- $ Install `requirements.txt`
```bash
$ pip install requirements.txt
```
- Run `crawler_main.py`
```bash
$ python crawler_main.py
```
Make sure you have an internet connection. By the end of the task, you can see the time consumed by all approaches individually.
#### **Some screenshots of the output:**
<img width="1022" alt="Screenshot 2022-10-17 at 3 15 30 PM" src="https://user-images.githubusercontent.com/55360312/196152519-89ae8d96-06c9-4448-9402-e5319420f072.png">

### **First run:**
<img width="477" alt="Screenshot 2022-10-17 at 2 52 49 PM" src="https://user-images.githubusercontent.com/55360312/196152570-810736e9-1dc4-4381-a5df-58a418442cb4.png">

### **Second run:**
<img width="405" alt="Screenshot 2022-10-17 at 3 06 06 PM" src="https://user-images.githubusercontent.com/55360312/196152580-8865a118-fe74-42e4-abf8-0582b0a5be92.png">
